enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,485 @@
+  controlPlaneEndpoint:
+    host: ""
+    port: 0
+  controlPlaneLoadBalancer:
+    additionalListeners:
+    - port: 22623
+      protocol: TCP
+    crossZoneLoadBalancing: false
+    loadBalancerType: nlb
+    name: capi-example-kbdpg-ext
+    scheme: internet-facing
+  network:
+    additionalControlPlaneIngressRules:
+    - description: MCS traffic from cluster network
+      fromPort: 22623
+      protocol: tcp
+      sourceSecurityGroupRoles:
+      - node
+      - controlplane
+      toPort: 22623
+    - description: controller-manager
+      fromPort: 10257
+      protocol: tcp
+      sourceSecurityGroupRoles:
+      - controlplane
+      - node
+      toPort: 10257
+    - description: kube-scheduler
+      fromPort: 10259
+      protocol: tcp
+      sourceSecurityGroupRoles:
+      - controlplane
+      - node
+      toPort: 10259
+    cni:
+      cniIngressRules:
+      - description: ICMP
+        fromPort: -1
+        protocol: icmp
+        toPort: -1
+       # several rules omitted for brevity
+      - description: Service node ports (UDP)
+        fromPort: 30000
+        protocol: udp
+        toPort: 32767
+    vpc:
+      availabilityZoneSelection: Ordered
+      availabilityZoneUsageLimit: 3
+      cidrBlock: 10.0.0.0/16
+  region: us-east-2
+  s3Bucket:
+    name: openshift-bootstrap-data-capi-example-kbdpg
+    presignedURLDuration: 1h0m0s
+status:
+  networkStatus:
+    apiServerElb:
+      attributes: {}
+  ready: false
+```
+
+This manifest describes the network, load balancers, security group rules, etc. that will be used or provisioned for
+the target cluster. Control plane machines are provisioned by [AWSMachines][AWSMachines] & [CAPI Machines][CAPIMachines]
+manifests.
+
+##### Additional Infrastructure
+
+The Installer will generate any additional infrastructure that is needed but not handled by the CAPI provider, either
+because it is out of scope (e.g. IAM, DNS) or not adopted upstream (e.g. split-horizon load balancers). The Installer
+will provide hooks into the provisioning lifecycle that can be used to provision resources using direct SDK calls
+or other tooling.
+
+The AWS proof-of-concept implementation utilizes hooks defined in this interface:
+
+```go
+type CAPIInfraHelper interface {
+	// PreProvision is called before provisioning using CAPI controllers has begun.
+	// and should be used to create dependencies needed for CAPI provisioning,
+	// such as IAM roles or policies.
+	PreProvision(in PreProvisionInput) error
+
+	// ControlPlaneAvailable is called once cluster.Spec.ControlPlaneEndpoint.IsValid()
+	// returns true, typically after load balancers have been provisioned. It can be used
+	// to create DNS records.
+	ControlPlaneAvailable(in ControlPlaneAvailableInput) error
+}
+
+type PreProvisionInput struct{ clusterID string }
+type ControlPlaneAvailableInput struct{ *clusterv1.Cluster }
+```
+
+For AWS, IAM roles needed by the CAPA provider are created with `PreProvision` and DNS Records
```

Comment:
can we clarify which DNS are records are these and how are they consumed?

Reply:
`api.` and `api-int.` as of today

Reply:
added